src/Microsoft.DotNet.Internal.DependencyInjection.Testing/DependencyInjectionValidation.cs

```diff
@@ -37,10 +37,13 @@ public static class DependencyInjectionValidation
         Action<ServiceCollection> register,
         out string errorMessage,
         IEnumerable<Type> additionalScopedTypes = null,
-        IEnumerable<Type> additionalSingletonTypes = null)
+        IEnumerable<Type> additionalSingletonTypes = null,
+        IEnumerable<string> additionalExemptTypes = null)
     {
         errorMessage = null;
 
+        var exemptTypes = s_exemptTypes.AddRange(additionalExemptTypes ?? Enumerable.Empty<string>());
```

Comment:
Style: It would be cleaner to initialize the variable to empty enumberable and then add range only if additionalExemptTypes is non-null (or add it always if the AddRange deals with null correctly).

Reply:
Also, why is AddRange result added to another variable when it's modifying the original one?

Reply:
`s_exemptTypes` is an immutable list, so when you call `AddRange` on it, it returns a new `ImmutableList`

Reply:
That being said, we should probably pass it as an immutable list too

Reply:
I improved it so we don't create a new list if there's no additionalExemptTypes